### PR TITLE
LEARNER-4195 Make sure jenkins automated jobs fail when error during the job

### PR DESCRIPTION
**Pull request by Ayub-Khan**
This pull request is to make sure jenkins sailthru and translation jobs fail when a error occur during the job process. The only was to do that is to raise exception and break the program.

### CODE REVIEW

File: edx_sailthru/sailthru_content/services/sailthru_api_service.py [hunk 1]
```diff
@@ -80,7 +82,7 @@ def clear(self):
                     response.json['error'],
                     response.json['errormsg']
                 )
-                return
+                raise RuntimeError('Failed to connect with Sailthru API')
```
Comment (amangano-edx) [inline edx_sailthru/sailthru_content/services/sailthru_api_service.py]: I'd recommend asking @schenedx to take a look at these sailthru changes, as I'm not familiar with these scripts.

File: transifex/utils.py [hunk 2]
```diff
@@ -297,7 +300,8 @@ def merge_pr(self):
                         'Failed to merge [%s/#%d], because of the exception [%s]',
                         self.name, self.pr.number, e,
                     )
-                    return False
+                    self.cleanup()
```
Comment (amangano-edx) [inline transifex/utils.py]: The cleanup method is called in a finally block, so I don't think you need to call it explicitly here, or anywhere else.

**Comment by amangano-edx** [comment]
Could you please clarify the purpose of these changes? Is it to get the builds to appear as 'failed' on jenkins so that it's easier to see when things go wrong?

File: edx_sailthru/sailthru_content/services/sailthru_api_service.py (hunk 1 above)
Comment (Ayub-Khan) [reply edx_sailthru/sailthru_content/services/sailthru_api_service.py]: sure.

**Comment by Ayub-Khan** [comment]
@amangano-edx yes the purpose is to get builds appear as 'failed'. Currently build shows success this way build will show failed if job failed and it would be easy to identify the problem. Sailthru job was failing for around 2 months and we had no idea because builds were showing success and issue was later identified when we checked courses were missing from feed. Although it was optional for translations because we get git notification too if a job failed but i moved forward by failing those too to make them similar and to represent job failure too.

File: transifex/utils.py [hunk 3]
```diff
@@ -297,7 +297,7 @@ def merge_pr(self):
                         'Failed to merge [%s/#%d], because of the exception [%s]',
                         self.name, self.pr.number, e,
                     )
-                    return False
+                    raise RuntimeError('Failed to merge.')
```
Comment (amangano-edx) [inline transifex/utils.py]: It might be good to specify that the failure was due to a GithubException, and maybe include the exception message?

**Comment by amangano-edx** [comment]
I think the translation script changes are probably fine, though there may be other places in the where they can fail but not raise an error. I'd recommend looking at 'push.py' and 'pull.py'.

I cannot speak to the Sailthru changes, as I'm not familiar with that code.

File: transifex/utils.py (hunk 3 above)
Comment (Ayub-Khan) [reply transifex/utils.py]: github exception is already being shown by logger.info so no need to show it again i think.

Comment (amangano-edx) [reply transifex/utils.py]: It might be good to log that at the error level instead of info, so it's clear in the log that it's an error.

**Comment by Ayub-Khan** [comment]
@amangano-edx  lets discuss this tonight

**Comment by Ayub-Khan** [comment]
Merging this.

**Pull request merged**